dragnet/features.py

```diff
@@ -97,47 +97,47 @@ class CSSFeatures(object):
     # appear in the CSS tags
-    attribute_tokens = {'id':['nav',
-                              'ss',
-                              'top',
-                              'content',
-                              'link',
-                              'title',
-                              'comment',
-                              'tools',
-                              'rating',
-                              'ss'],
-                     'class':['menu',
-                              'widget',
-                              'nav',
-                              'share',
-                              'facebook',
-                              'cat',
-                              'top',
-                              'content',
-                              'item',
-                              'twitter',
-                              'button',
-                              'title',
-                              'header',
-                              'ss',
-                              'post',
-                              'comment',
-                              'meta',
-                              'alt',
-                              'time',
-                              'depth',
-                              'thread',
-                              'author',
-                              'tools',
-                              'reply'
-                              'url',
-                              'avatar',
-                              'ss']}
+    attribute_tokens = {'id': ['nav',
+                               'ss',
+                               'top',
+                               'content',
+                               'link',
+                               'title',
+                               'comment',
+                               'tools',
+                               'rating',
+                               'ss'],
+                        'class': ['menu',
+                                  'widget',
+                                  'nav',
+                                  'share',
+                                  'facebook',
+                                  'cat',
+                                  'top',
+                                  'content',
+                                  'item',
+                                  'twitter',
+                                  'button',
+                                  'title',
+                                  'header',
+                                  'ss',
+                                  'post',
+                                  'comment',
+                                  'meta',
+                                  'alt',
+                                  'time',
+                                  'depth',
+                                  'thread',
+                                  'author',
+                                  'tools',
+                                  'reply',
```

Comment:
Ah this is the problem here -- in your edit, the `'reply'` (correctly) has a training comma.  In the original code, it does not which implicitly concatenates it with the following line `'url'`.  This explains both strange results: that leaving in the duplicate '`ss`' changes the dimensionality of the features vector and the failing unit test.

If you remove this comma (or more explicitly create a single entry `'reply' + 'url'` with a comment explaining the strange entry) and add back in the duplicate `ss` two lines lower the tests should pass.